### ISSUE TITLE
Why add 1e-4 in softmax

### ISSUE THREAD

**Issue by Andy1621** [jayleicn/singularity#27]
Hi! I found that `1e-4` is added in the `softmax` function. Are there any reasons?
https://github.com/jayleicn/singularity/blob/bf4a86ec7506565d1f6805ee1612aa6029592776/models/model_retrieval_base.py#L303-L304

If it is used for smoothing, why not add it after `softmax`, like `F.softmax(sim_i2t, dim=1)+1e-4`.

**Comment by jayleicn** [comment]
To avoid zero-division error in F.softmax.

**Comment by Andy1621** [comment]
<img width="228" alt="image" src="https://user-images.githubusercontent.com/24236723/226086429-c544aad8-2b61-4efe-a6a4-2369c2a695f8.png">

The softmax uses `exp`. It seems that it will not cause zero-division error.

**Comment by jayleicn** [comment]
Oh, you are right! I was looking at it too quickly. Please see this issue https://github.com/salesforce/ALBEF/issues/93

**Comment by Andy1621** [comment]
Thanks!
